### DISCUSSION TITLE
Binding to a date attribute in an Eloquent Model when in a different timezone than UTC

### DISCUSSION THREAD

**Discussion by oxytoxin** [livewire/livewire#3212] [7. Bugs (archived)]
I have an eloquent model called Task that has a deadline column which is included in the model's $dates attribute array. My timezone in the config files is set to Asia/Manila.
I have a livewire component that binds to this Task model using wire:model. The problem I encountered was that everytime a livewire action gets called, e.g. via a button click that triggers an action, the model's deadline attribute gets readjusted.

To be more precise, it seems that the deadline's value is taken as having a UTC timezone and gets readjusted to my set timezone in the config files every time livewire makes a roundtrip. Does this perhaps have something to do with the model's Serialization & Deserialization?

![image](https://user-images.githubusercontent.com/61694153/124005397-9ab54b00-da0b-11eb-9163-c798e590ac1e.png)
![image](https://user-images.githubusercontent.com/61694153/124005471-b4569280-da0b-11eb-91e1-289530574630.png)

This is what it looks like before pressing a button
![image](https://user-images.githubusercontent.com/61694153/124005712-f97ac480-da0b-11eb-9d15-4564f0342f71.png)
And after pressing validate
![image](https://user-images.githubusercontent.com/61694153/124005753-07304a00-da0c-11eb-813f-a08ced6bd753.png)

Thank you for your help! Cheers!

**Comment by potsky** [comment]
Exactly the same problem, it seems that Livewire do not hydrate dates correctly @calebporzio !

The workaround for us is to re-query the database on every Livewire request to get fresh models instead of relying on Livewire objects :-(

**Reply by samlev** [reply]
This initial discussion would have been for Livewire v2. It's likely not to do with Livewire serializing dates at all - I believe that v2 was using Laravel's serializer/deserializer, and it would have either converted dates to a timestamp (which doesn't contain timezone information), or to a date-time string that, but default, also doesn't contain timezone information.

In v3, models don't get serialized at all unless you turn on legacy model binding (which you should avoid)
